internal/store/driver.go

```
@@ -642,8 +844,424 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
+								}
+
+							}
+							// set different customMetadata for gateways next
+							route.Metadata = d.customMetadata
+
+							edge.Spec.Routes = append(edge.Spec.Routes, route)
+						}
+					}
+				}
+			}
+
+			edgeMap[domainName] = edge
+		}
+	}
+}
+
+type Actions struct {
+	endpointActions []ingressv1alpha1.EndpointAction
+}
+
+func (d *Driver) createEndpointPolicyForGateway(rule *gatewayv1.HTTPRouteRule) (*ingressv1alpha1.EndpointPolicy, error) {
+	inboundActions := Actions{}
+	outboundActions := Actions{}
+	expressions := []string{}
+	pathPrefixMatches := []string{}
+
+	// NOTE: matches are only defined on requests, and fitlers are only triggered by matches,
+	// but some fitlers define transformations on responses, so we need to define matches on both
+	// Policy.Inbound and Policy.Outbound when possible to work with ngrok's system
+	for _, match := range rule.Matches {
+		if match.Path != nil {
+			if match.Path.Type != nil {
+				switch *match.Path.Type {
+				case gatewayv1.PathMatchExact:
+				case gatewayv1.PathMatchPathPrefix:
+					if match.Path.Value != nil {
+						pathPrefixMatches = append(pathPrefixMatches, *match.Path.Value)
+					}
+				case gatewayv1.PathMatchRegularExpression:
+					return nil, errors.NewErrorNotFound(fmt.Sprintf("unsupported match type PathMatchType %v found", *match.Path.Type))
+				default:
+					return nil, errors.NewErrorNotFound(fmt.Sprintf("Unknown match type PathMatchType %v found", *match.Path.Type))
+				}
+			}
+		}
+
+		if match.Method != nil {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPMethod", *match.Method)
+		}
+
+		if len(match.Headers) > 0 {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPHeaderMatch", match.Headers)
+		}
+
+		if len(match.QueryParams) > 0 {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPQueryParamMatch", match.QueryParams)
```

should there return errors here? (and above)

---

yes, agreed, it should throw an error here.